src/components/pages/doc.js

```diff
@@ -157,7 +157,7 @@ function Doc({ initialState, PageTemplate }) {
   const isEditingDisabled = state.currentDocOrRevision.archived || isExternalDocument || state.type !== DOCUMENT_TYPE.document;
   const isViewRevisionsButtonVisible = state.type === DOCUMENT_TYPE.document;
   const isViewDocumentButtonVisible = state.type === DOCUMENT_TYPE.revision;
-  const isHardDeleteEnabled = state.type === DOCUMENT_TYPE.revision;
+  const disabledSectionActions = [...isExternalDocument ? [SECTION_ACTIONS.hardDelete] : []];
```

Comment:
This construct seems backwards to me. I had to dig down 3 files from here to understand why you send "disabledActions" instead of "possible/enabledActions".

Reply:
This also changes the fact that you can only hard delete when you see revisions. You can now delete on the document view, is this wanted? (I don't think it's a bad idea).

Reply:
`disabledActions` is because that is the exception from the rule, and the rule is that all actions are enabled.

Reply:
I will check your second remark/comment here, though unless I'm missing something I don't get why it would behave like you are mentioning 🤔

Reply:
Try the same on staging. You don't see the "delete section" when you just view the document. If you look at the line above that is the difference
  const isHardDeleteEnabled = state.type === DOCUMENT_TYPE.revision
I think this got lost

Reply:
Ok I checked and also better understood what you meant. The fact that you can **hard delete** a section from a revision is, from what I know, actually what is wanted. The reason for this functionality is to be able to delete content that should legally not be on the website, in whichever revision it is (not only the last). That's also why the dialog when you delete asks you if you want to delete the same section in all other revisions (backwards and forwards basically)

Reply:
I don't think you understood what I meant :), the bug is still there.

Reply:
"Try the same on staging. You don't see the "delete section" when you just view the document"
I think this is something you introduced on staging with the previous ticket but that is a bug, it basically broke the intended functionality.
@ahelmberger , what's your take on this?

Reply:
This wasn't a bug, it was because previously when you came to the doc.js you didn't see the "article" (aka the document) but rather all revisions (like you have when you now click the "view revisions"). Think of the doc.js page as the article page when you come to it the first time :) and try that on prod.

Reply:
If you go on prod now, on a document page as is (view mode, not edit mode) and hover on any section (in any revision), you see the option to hard delete the section.

Reply:
> `disabledActions` is because that is the exception from the rule, and the rule is that all actions are enabled.

To me it seems weird that the logic for what is enables / disables the actions now lives in 2 places:
 doc.js has:   const disabledSectionActions = [...isExternalDocument ? [SECTION_ACTIONS.hardDelete] : []];
 section-display:  in between that just passes the disabledActions 
 section-action-dropdown: again knows about all actions and adds some additional restraints on this particular action:
if (canHardDelete && !section.deletedOn && //the rest)

Would it not be more streamlined if the 
   section-action-dropdown just knew how to display in stuff as a dumb component that receives it's actions from someone who knows? 
Alternatively, it could keep all the conditions. It's the split that is a bit hard to maintain I think.

Reply:
> If you go on prod now, on a document page as is (view mode, not edit mode) and hover on any section (in any revision), you see the option to hard delete the section.

And if you go to the articles page (which was merged in the doc page) you see you can't delete it :)

Reply:
In the former article page there was no option to delete a section, only in the docs view (with the revision slider), and @laurentiu-ilici kept this behavior also for the now combined pages. Now with this change also the article view mode allows deletions. I don't really have an opinion on what would be better.

Reply:
But it couldn't hurt to also be able to delete a section without going into revisions mode before, right?

Reply:
> If you go on prod now, on a document page as is (view mode, not edit mode) and hover on any section (in any revision), you see the option to hard delete the section.

Anyway, as I said we can keep this new way of doing things, but then the code needs to change a bit. If you now delete all revisions, you get a blank screen after saying "Ja". Then you need to refresh the page to get to a stable state.

Reply:
Regarding hard deletion in various view modes, I will make the smallest change for now and allow hard-delete only for document view mode.

Reply:
You mean for revision view mode :)). I see this is very confusing. Okay, we will improve in a different ticket, the important thing is that it still works.